crypto/txscript/src/opcodes/mod.rs

```diff
@@ -2973,4 +3004,376 @@ mod test {
             TestCase { code: opcodes::OpIfDup::empty().expect("Should accept empty"), init: vec![vec![]], dstack: vec![vec![]] },
         ])
     }
+
+    mod kip10 {
```

Comment:
These tests are fine (you can leave them), but I expected tests with the context of the whole script, where you execute the whole script engine, and not just the opcode (like you did in the examples).

These tests should focus on very simple cases, like redeeming a UTXO with script `0 OP_OUTPUTVALUE 100 EQUAL`, and try to redeem it with tx with 100 sompi on the first input, and another value (and similar tests for the other new op codes)

Reply:
added by commit 5689ed5d01212719910e90d369806313aca8ab78

OpInputSpk can't be tested this way since catch22. Script needs to know spk of itself, spk depends on the script